GVRf/Framework/framework/src/main/jni/objects/mesh.cpp

```diff
@@ -208,8 +211,159 @@ void Mesh::bindVertexAttributes(GLuint programID)
     unSetVaoDirty();
 }
 
+void Mesh::createAttributeMapping(int programId,
+        int& totalStride, int& attrLen)
+{
+    dumpActiveAttribues(programId);
+    totalStride = attrLen = 0;
+    if (programId == -1)
+    {
+        // If program id has not been set, return.
+        return;
+    }
+    GLint numActiveAtributes;
+    glGetProgramiv(programId, GL_ACTIVE_ATTRIBUTES, &numActiveAtributes);
+    GLchar attrName[512];
+    GLAttributeMapping attrData;
+
+    for (int i = 0; i < numActiveAtributes; i++)
+    {
+        GLsizei length;
+        GLint size;
+        GLenum type;
+        glGetActiveAttrib(programId, i, 512, &length, &size, &type, attrName);
+        if (std::find(dynamicAttribute_Names_.begin(), dynamicAttribute_Names_.end(), attrName) != dynamicAttribute_Names_.end())
+        {
+            // Skip dynamic attributes. Currently only bones are dynamic attributes which changes each frame.
+            // They are handled seperately.
+        }
+        else
+        {
+            attrData.type = GL_FLOAT;
+            attrData.index = i;
+            attrData.data = NULL;
+            attrData.offset = totalStride;
+            bool addData = true;
+            int len = 0;
+
+            // Two things to note --
+            // 1. The 3 builtin buffers are still seperate from the maps used for the other attributes
+            // 2. The attribute index *has* to be 0, 1 and 2 for position, tex_coords and normal. The
+            // index from querying via glGetActiveAttrib cannot be used. Needs analysis.
+            if (strcmp(attrName, "a_position") == 0)
+            {
+                attrData.size = 3;
+                len = vertices_.size();
+                attrData.data = vertices_.data();
+                attrData.index = 0;
+            }
+            else if (strcmp(attrName, "a_normal") == 0)
+            {
+                attrData.size = 3;
+                len = normals_.size();
+                attrData.data = normals_.data();
+                attrData.index = 2;
+            }
+            else if (strcmp(attrName, "a_texcoord") == 0)
+            {
+                attrData.size = 2;
+                len = tex_coords_.size();
+                attrData.data = tex_coords_.data();
+                attrData.index = 1;
+            }
+            else if (strcmp(attrName, "a_tex_coord") == 0)
```

Comment:
In your future commit please put all these magic names in some header and lets start using symbolic constants already.